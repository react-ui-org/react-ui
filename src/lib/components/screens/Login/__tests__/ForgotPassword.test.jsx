import React from 'react';
import { mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import sinon from 'sinon';
import { ForgotPassword } from '..';

describe('rendering', () => {
  it('renders correctly', () => {
    const tree = mount(<ForgotPassword />);

    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('renders correctly with all props except translations', () => {
    const tree = mount(<ForgotPassword
      id="custom-id"
      logo="http://satyr.io/100x100/33?text=logo"
      title="Company"
      error="Error message"
      footer={(
        <a href="http://example.com">link</a>
      )}
      usernameType="username"
    />);

    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('renders correctly', () => {
    const tree = mount(<ForgotPassword
      translations={{
        email: 'E-mail',
        resetPassword: 'Reset password',
      }}
    />);

    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});

describe('functionality', () => {
  it('calls submitHandler() on submit', () => {
    const spy = sinon.spy();
    const component = mount(<ForgotPassword submitHandler={spy} />);

    component
      .find('form')
      .simulate('submit');
    expect(spy.calledOnce).toEqual(true);
  });
});
