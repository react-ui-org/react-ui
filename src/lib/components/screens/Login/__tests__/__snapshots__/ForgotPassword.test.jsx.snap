// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders correctly 1`] = `
<WithTranslationContextComponent
  translations={null}
>
  <ForgotPassword
    error={null}
    footer={null}
    logoUrl={null}
    onChangeHandler={null}
    submitHandler={null}
    title={null}
    translations={
      Object {
        "email": "E-mail",
        "resetPassword": "Reset password",
      }
    }
    usernameType="email"
  >
    <div
      className="root"
    >
      <div
        className="box"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="mb-3"
          >
            <ForwardRef(withForwardedRef(TextField))
              autoComplete="username"
              changeHandler={[Function]}
              fullWidth={true}
              id="resetEmailInput"
              label="E-mail"
              required={true}
              type="email"
            >
              <TextField
                autoComplete="username"
                changeHandler={[Function]}
                disabled={false}
                forwardedRef={null}
                fullWidth={true}
                helperText={null}
                id="resetEmailInput"
                inputSize={null}
                isLabelVisible={true}
                label="E-mail"
                layout="vertical"
                placeholder={null}
                required={true}
                size="medium"
                type="email"
                validationState={null}
                variant="outline"
              >
                <label
                  className="root
        isRootFullWidth
        rootLayoutVertical
        isRootRequired
        rootSizeMedium
        
        
        rootVariantOutline"
                  htmlFor="resetEmailInput"
                  id="resetEmailInput__label"
                >
                  <div
                    className="label"
                    id="resetEmailInput__labelText"
                  >
                    E-mail
                  </div>
                  <div
                    className="inputContainer"
                  >
                    <input
                      autoComplete="username"
                      className="input"
                      disabled={false}
                      id="resetEmailInput"
                      onChange={[Function]}
                      placeholder={null}
                      required={true}
                      size={null}
                      type="email"
                    />
                  </div>
                </label>
              </TextField>
            </ForwardRef(withForwardedRef(TextField))>
          </div>
          <ForwardRef(withForwardedRef(Button))
            block={true}
            label="Reset password"
            type="submit"
          >
            <Button
              afterLabel={null}
              beforeLabel={null}
              block={true}
              clickHandler={null}
              disabled={false}
              endCorner={null}
              forwardedRef={null}
              grouped={false}
              label="Reset password"
              labelVisibility="all"
              loadingIcon={null}
              priority="default"
              size="medium"
              startCorner={null}
              type="submit"
              variant="primary"
            >
              <button
                className="root
        priorityDefault
        sizeMedium
        variantPrimary
        isRootBlock"
                disabled={false}
                onClick={null}
                title={null}
                type="submit"
              >
                <span
                  className="label"
                >
                  Reset password
                </span>
              </button>
            </Button>
          </ForwardRef(withForwardedRef(Button))>
        </form>
      </div>
    </div>
  </ForgotPassword>
</WithTranslationContextComponent>
`;

exports[`rendering renders correctly 2`] = `
<WithTranslationContextComponent
  translations={
    Object {
      "email": "E-mail",
      "resetPassword": "Reset password",
    }
  }
>
  <ForgotPassword
    error={null}
    footer={null}
    logoUrl={null}
    onChangeHandler={null}
    submitHandler={null}
    title={null}
    translations={
      Object {
        "email": "E-mail",
        "resetPassword": "Reset password",
      }
    }
    usernameType="email"
  >
    <div
      className="root"
    >
      <div
        className="box"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="mb-3"
          >
            <ForwardRef(withForwardedRef(TextField))
              autoComplete="username"
              changeHandler={[Function]}
              fullWidth={true}
              id="resetEmailInput"
              label="E-mail"
              required={true}
              type="email"
            >
              <TextField
                autoComplete="username"
                changeHandler={[Function]}
                disabled={false}
                forwardedRef={null}
                fullWidth={true}
                helperText={null}
                id="resetEmailInput"
                inputSize={null}
                isLabelVisible={true}
                label="E-mail"
                layout="vertical"
                placeholder={null}
                required={true}
                size="medium"
                type="email"
                validationState={null}
                variant="outline"
              >
                <label
                  className="root
        isRootFullWidth
        rootLayoutVertical
        isRootRequired
        rootSizeMedium
        
        
        rootVariantOutline"
                  htmlFor="resetEmailInput"
                  id="resetEmailInput__label"
                >
                  <div
                    className="label"
                    id="resetEmailInput__labelText"
                  >
                    E-mail
                  </div>
                  <div
                    className="inputContainer"
                  >
                    <input
                      autoComplete="username"
                      className="input"
                      disabled={false}
                      id="resetEmailInput"
                      onChange={[Function]}
                      placeholder={null}
                      required={true}
                      size={null}
                      type="email"
                    />
                  </div>
                </label>
              </TextField>
            </ForwardRef(withForwardedRef(TextField))>
          </div>
          <ForwardRef(withForwardedRef(Button))
            block={true}
            label="Reset password"
            type="submit"
          >
            <Button
              afterLabel={null}
              beforeLabel={null}
              block={true}
              clickHandler={null}
              disabled={false}
              endCorner={null}
              forwardedRef={null}
              grouped={false}
              label="Reset password"
              labelVisibility="all"
              loadingIcon={null}
              priority="default"
              size="medium"
              startCorner={null}
              type="submit"
              variant="primary"
            >
              <button
                className="root
        priorityDefault
        sizeMedium
        variantPrimary
        isRootBlock"
                disabled={false}
                onClick={null}
                title={null}
                type="submit"
              >
                <span
                  className="label"
                >
                  Reset password
                </span>
              </button>
            </Button>
          </ForwardRef(withForwardedRef(Button))>
        </form>
      </div>
    </div>
  </ForgotPassword>
</WithTranslationContextComponent>
`;

exports[`rendering renders correctly with all props except translations 1`] = `
<WithTranslationContextComponent
  error="Error message"
  footer={
    <a
      href="http://example.com"
    >
      link
    </a>
  }
  id="custom-id"
  logo="http://satyr.io/100x100/33?text=logo"
  title="Company"
  translations={null}
  usernameType="username"
>
  <ForgotPassword
    error="Error message"
    footer={
      <a
        href="http://example.com"
      >
        link
      </a>
    }
    id="custom-id"
    logo="http://satyr.io/100x100/33?text=logo"
    logoUrl={null}
    onChangeHandler={null}
    submitHandler={null}
    title="Company"
    translations={
      Object {
        "email": "E-mail",
        "resetPassword": "Reset password",
      }
    }
    usernameType="username"
  >
    <div
      className="root"
      id="custom-id"
    >
      <div
        className="title"
        id="custom-id__title"
      >
        Company
      </div>
      <div
        className="box"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="error"
            id="custom-id__errorText"
          >
            Error message
          </div>
          <div
            className="mb-3"
          >
            <ForwardRef(withForwardedRef(TextField))
              autoComplete="username"
              changeHandler={[Function]}
              fullWidth={true}
              id="custom-id__resetEmailInput"
              label="E-mail"
              required={true}
              type="text"
            >
              <TextField
                autoComplete="username"
                changeHandler={[Function]}
                disabled={false}
                forwardedRef={null}
                fullWidth={true}
                helperText={null}
                id="custom-id__resetEmailInput"
                inputSize={null}
                isLabelVisible={true}
                label="E-mail"
                layout="vertical"
                placeholder={null}
                required={true}
                size="medium"
                type="text"
                validationState={null}
                variant="outline"
              >
                <label
                  className="root
        isRootFullWidth
        rootLayoutVertical
        isRootRequired
        rootSizeMedium
        
        
        rootVariantOutline"
                  htmlFor="custom-id__resetEmailInput"
                  id="custom-id__resetEmailInput__label"
                >
                  <div
                    className="label"
                    id="custom-id__resetEmailInput__labelText"
                  >
                    E-mail
                  </div>
                  <div
                    className="inputContainer"
                  >
                    <input
                      autoComplete="username"
                      className="input"
                      disabled={false}
                      id="custom-id__resetEmailInput"
                      onChange={[Function]}
                      placeholder={null}
                      required={true}
                      size={null}
                      type="text"
                    />
                  </div>
                </label>
              </TextField>
            </ForwardRef(withForwardedRef(TextField))>
          </div>
          <ForwardRef(withForwardedRef(Button))
            block={true}
            id="custom-id__resetPasswordButton"
            label="Reset password"
            type="submit"
          >
            <Button
              afterLabel={null}
              beforeLabel={null}
              block={true}
              clickHandler={null}
              disabled={false}
              endCorner={null}
              forwardedRef={null}
              grouped={false}
              id="custom-id__resetPasswordButton"
              label="Reset password"
              labelVisibility="all"
              loadingIcon={null}
              priority="default"
              size="medium"
              startCorner={null}
              type="submit"
              variant="primary"
            >
              <button
                className="root
        priorityDefault
        sizeMedium
        variantPrimary
        isRootBlock"
                disabled={false}
                id="custom-id__resetPasswordButton"
                onClick={null}
                title={null}
                type="submit"
              >
                <span
                  className="label"
                  id="custom-id__resetPasswordButton__labelText"
                >
                  Reset password
                </span>
              </button>
            </Button>
          </ForwardRef(withForwardedRef(Button))>
        </form>
        <div
          className="footer"
          id="custom-id__footerContent"
        >
          <a
            href="http://example.com"
          >
            link
          </a>
        </div>
      </div>
    </div>
  </ForgotPassword>
</WithTranslationContextComponent>
`;
